### ISSUE TITLE
Ajax response with status 0 must throw an error

### ISSUE THREAD

**Issue by eltiodelsofa** [ReactiveX/rxjs#3606]
If ajax observable has parameter timeout configured, with a slow request, ajax is emitting a next instead of an error.

**RxJS version:**
 6.0.0

**Code to reproduce:**
```js
ajax({
  url: 'https://reqres.in/api/users?delay=5',
  timeout: 500,
}).subscribe({
  next: v => console.log('next', v),
  error: err => console.log('err', err),
});
```

**Expected behavior:**

Emit an error event with an AjaxError object

**Actual behavior:**

Emit a next event with ajax null response and status 0

**Comment by felixfbecker** [comment]
Workaround:
```ts
tap(response => {
            if (response.status === 0) {
                throw Object.assign(new Error('Ajax status 0'), response)
            }
        }),
```

**Comment by adriensamson** [comment]
I believe #3653 also fixes the issue with down network: moving the next() logic from `onreadystatechange` to `onload` allows `onerror` to trigger first.

**Comment by felixfbecker** [comment]
@bbonnet Was this fixed by https://github.com/ReactiveX/rxjs/pull/3653?

**Comment by goenning** [comment]
Tested and working on 6.4.*

![rxjs-ajax](https://user-images.githubusercontent.com/94755/57067684-95db9c00-6cc7-11e9-8f7c-6c303f005093.PNG)

**Comment by ndlarsen** [comment]
In which version was this actually fixed?

**Comment by cartant** [comment]
Closed by #3653
